### ISSUE TITLE
the specific implementation process of the AdaNeg method

### ISSUE THREAD

**Issue by shuaiNJU** [YBZh/OpenOOD-VLM#8]
I would like to ask in which.py file the specific implementation process of the AdaNeg method is located?

Thanks a lot！

**Comment by shuaiNJU** [comment]
Thanks a lot！
I have another issue.  In  openood\evaluators\fsood_evaluator_clip.py (line 369):
```
combined_dataset = ConcatDataset([id_loader.dataset, ood_dl.dataset])
combined_dataloader = DataLoader(combined_dataset, batch_size=id_loader.batch_size, num_workers=id_loader.num_workers, shuffle=True)
```
a combiend_dataset is used to inference,  but how in-distribution scores and ood scores are defined for the combiend_dataset.  I see that you do not distinguish these two scores   rather than using "conf" in  line 392 :
`ood_metrics = compute_all_metrics(**conf**, label, pred)`

In addition, tha label number used there is 0~999?

**Comment by YBZh** [comment]
For test-time adaptation methods including AdaNeg, we structure the test stream to contain both in-distribution (ID) and out-of-distribution (OOD) samples, better reflecting real-world scenarios. 

As for the labels, following standard conventions, we assign label indices 0-999 to ID data and -1 to OOD data.
